src/lib/arcgis.js

```diff
@@ -57,13 +56,32 @@ async function getServices (url) {
   }
 }
 
+/**
+ * An ArcGIS service endpoint with accessors for its parts
+ *
+ * @type       {class}
+ */
 var ArcGISService = class ArcGISService {
+  /**
+   * Requires an url to the root of the service, such as one returned from the
+   * `getServices()` call.
+   *
+   * @param      {String}  url     The url to the REST endpoint
+   */
   constructor (url) {
     this.baseUrl = url
   }
 
+  /**
+   * Return a list of the layers available
+   *
+   * @return     {Promise}  Eventually returns a list of `ArcGISLayer` objects
```

Comment:
Thanks for adding this documentation.

LMK if you want to talk about adding Flow or TypeScript to this project and convert these comments to something enforceable.

Reply:
I think that needs to be a ticket in and of itself.